CommandLine/testing/TestHarness.hpp

```diff
@@ -61,11 +61,11 @@ class TestHarness {
   /// Launch a game's executable file and "attach" to it.
   /// Not in a debugger sense, but in a "grabbing the window" sense.
   static std::unique_ptr<TestHarness>
-      launch_and_attach(const std::string &game, const TestConfig &tc);
+      launch_and_attach(const std::string &game, TestConfig &tc);
```

Comment:
Style rule: If you intend to modify a parameter, have it be passed as a pointer rather than a reference. Only use const references. This is a fantastic example of this rule, because I have no idea why `launch_and_attach` should ever be modifying the test config I pass it. But at least if I have to say & on my side, I'll recognize that it is.

Reply:
This helped me begin understanding why C++ forbids passing temporaries for non-const reference parameters. I will change the parameters to non-const pointers instead. `launch_and_attach` needs to mutate the `workdir` member of `TestConfig` so that `gather_coverage(...)` knows where to tell `lcov` to look for coverage data.

Reply:
That's also fine.

Reply:
Alright, I changed `build_game(...)` to take a pointer to the TestConfig since it mutates it: f5243a72190a77f019c0aef061668c168f942d2b

`launch_and_attach` and `run_to_completion` can now keep the const qualification on their TestConfig parameters, making both of them safe to pass a temporary to (and allowing me to undo earlier changes that are now unnecessary).